lib/object.h

```diff
@@ -13,6 +13,44 @@ struct tpm2_loaded_object {
     tpm2_session *session;
 };
 
+struct tpm_tool_ctx {
+    /*
+     * Inputs
+     */
+    struct {
+        const char *ctx_path;
+        const char *auth_str;
+        tpm2_loaded_object object;
+    } parent;
+
+    struct {
+        const char *pubpath;
+        TPM2B_PUBLIC public;
+        const char *privpath;
+        TPM2B_PRIVATE private;
+        ESYS_TR handle;
+    } object;
+
+    /*
+     * Outputs
+     */
+    const char *namepath;
+    const char *contextpath;
+
+    /*
+     * Parameter hashes
+     */
+    const char *cp_hash_path;
+    TPM2B_DIGEST cp_hash;
+    bool is_command_dispatch;
+    TPMI_ALG_HASH parameter_hash_algorithm;
+
+    /*
+     * PEM specific
+     */
+    bool is_pem;
```

Comment:
These do not belong in object.h. They should be in the tpm2_load.c

Reply:
Made these changes with the latest push